### PR TITLE
Add research gate

### PR DESCRIPTION
**Pull request by etienne428**
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        String link = eLink.select("a[href^=https]").select("a[href$=.pdf]").attr("href");
+        LOGGER.debug("PDF link: {}", link);
+
+        if (link.contains("researchgate.net")) {
+            return Optional.of(new URL(link));
+        }
+        return Optional.empty();
+    }
+
+    String getURLByString(String query) throws IOException, FetcherException, NullPointerException {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java:111]: private

Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java:111]:
```suggestion
    private String getURLByString(String query) throws IOException, FetcherException, NullPointerException {
```

File: src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                                 .get();
+
+            link = HOST + Objects.requireNonNull(html.getElementById("content"))
+                                 .select("a[href^=publication/]")
+                                 .attr("href");
+            if (link.contains("?")) {
+                link = link.substring(0, link.indexOf("?"));
+            }
+        } catch (URISyntaxException e) {
+            return null;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java:134]: Not happy about returning null here. Maybe empty string?

File: src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java (hunk 1 above)
Comment (etienne428) [reply src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java:111]: Making it private means I can't call it from the test class... Would "protected" be better?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java:111]: Ah did not see that you have test for it. Then leave it as is

File: src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java (hunk 2 above)
Comment (etienne428) [reply src/main/java/org/jabref/logic/importer/fetcher/ResearchGate.java]: I refactored the Code to return Optional<String>

**Review by Siedlerchr — APPROVED** [review]

File: src/test/resources/tinylog-test.properties [hunk 3]
```diff
@@ -1,4 +1,3 @@
 level = info
 writer = console
-# uncomment the following line to enable debug logging outputs
-#level@org.jabref.model.entry.BibEntry = debug
+level@org.jabref.logic.importer.fetcher.ResearchGate = trace
```
Comment (Siedlerchr) [inline src/test/resources/tinylog-test.properties:3]:
```suggestion
# uncomment the following line to enable debug logging outputs
# level@org.jabref.logic.importer.fetcher.ResearchGate = trace
```
Please restore the comments, so we can enable this only if needed

**Review by calixtus — APPROVED** [review]
Made some small changes to comments. Looks good to me!

File: src/test/java/org/jabref/logic/importer/fetcher/ResearchGateTest.java [hunk 4]
```diff
@@ -119,6 +119,7 @@ void performSearchWithBibEntry() throws FetcherException {
     }
 
     @Test
+    @DisabledOnCIServer("CI server is unreliable")
     void performSearchWithTitleWithCurlyBraces() throws FetcherException {
         BibEntry entryInput = new BibEntry()
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/fetcher/ResearchGateTest.java:124]: Easier if you also set the StandardEntryType Misc here and then in the assertEquals just compare both entries

**Pull request merged**